scripts/run_soupX.R

```diff
@@ -45,20 +46,134 @@ clusters_file <- opts$cluster
 sample_name <- opts$sample_name
 
 # load 10X data and convert to Soup Channel (object SoupX uses for analysis)
-fil_mat <- Read10X_h5(fil, use.names = T)
-raw_mat <- Read10X_h5(raw, use.names = T)
+filtered_matrix <- Read10X_h5(fil, use.names = TRUE)
+raw_matrix <- Read10X_h5(raw, use.names = TRUE)
+
+############################################################################
+# If cluster file available
 clusters <- read.csv(clusters_file)
-mDat <- data.frame(clusters=clusters$Cluster,row.names=clusters$Barcode)
-sc <- SoupChannel(raw_mat, fil_mat, mDat)
+mDat <- data.frame(clusters=clusters$Cluster, row.names=clusters$Barcode)
+sc <- SoupChannel(raw_matrix, filtered_matrix, mDat)
+
+############################################################################
+# If cluster file NOT available
+# Make a Seurat object from the sparce matrix
+seurat_obj  <- CreateSeuratObject(counts = filtered_matrix)
```

Comment:
Is there a flag to create the clusters file if none is available? Otherwise, it seems regardless if one is provided, it will create clusters anyway